drivers/sensor/lis2dh/lis2dh.h

```diff
@@ -15,27 +15,22 @@
 #include <drivers/sensor.h>
 #include <string.h>
 
-#define LIS2DH_BUS_ADDRESS		DT_INST_REG_ADDR(0)
-#define LIS2DH_BUS_DEV_NAME		DT_INST_BUS_LABEL(0)
+#define DT_DRV_COMPAT st_lis2dh
```

Comment:
Is it adviced to set DT_DRV_COMPAT in header file?
I already have some issues trying to set it in some header files so I am wondering it is a correct manner to do.

Reply:
Well, actually before it was in the spi and i2c sensor driver .c file, but I moved it in .h as it is common to both. But it is a valid point. It seems I'm the only one to define it in a header file. I can move it back to the .c files.
@mbolivar-nordic any feedback here?

Reply:
I don't think it's a good idea in general to have it in a header file because you likely want that macro defined only for the particular .c file where you are using DT macros. You could also surround it with `#ifndef DT_DRV_COMPAT` I guess, but I personally think it's cleaner to move it to the .c file, although I will defer to @mbolivar-nordic.

Reply:
@carlescufi 
OK, clear! I'll do a quick change right now!
I will also incorporate the change requested by @mbolivar-nordic here above

Reply:
> Is it adviced to set DT_DRV_COMPAT in header file?

Looks like this is just an internal header used as shared code by the driver implementation, so I don't think there's any reason not to define DT_DRV_COMPAT here. There is no policy for or against this that I am aware of.